BinaryEncoding.md

```diff
@@ -1,122 +1,398 @@
-
-* A module contains (in this order):
-  - A header, containing:
-    + The [magic number](https://en.wikipedia.org/wiki/Magic_number_%28programming%29)
-    + Other data TBD
-  - A table (sorted by offset) containing, for each section:
-    + A string literal section type name
-    + 64-bit offset within the module
-  - A sequence of sections
-* A section contains:
-  - A header followed by
-  - The section contents (specific to the section type)
-* A `definitions` section contains (in this order):
-  - The generic section header
-  - A table (sorted by offset) containing, for each type which has operators:
-    + A standardized string literal [type name](AstSemantics.md#expression-types).
-      The index of a type name in this table is referred to as a type ID
-    + 64-bit offset of its operator table within the section
-  - A sequence of operator tables
-  - An operator table contains:
-    + A sequence of standardized string literal [operator names](AstSemantics.md),
-      where order determines operator index
-* A `code` section contains (in this order):
-  - The generic section header
-  - A table (sorted by offset) containing, for each function:
-     + Signature
-     + Function attributes, valid attributes TBD (could include hot/cold, optimization level, noreturn, read/write/pure, ...)
-     + 64-bit offset within the section
-  - A sequence of functions
-  - A function contains:
-    + A table containing, for each type ID that has [locals](AstSemantics.md#local-variables):
-      * Type ID
-      * Count of locals
-    + The serialized AST
-* A `data` section contains (in this order):
-   - The generic section header
-   - A sequence of byte ranges within the binary and corresponding addresses in the linear memory
-
-
-All strings are encoded as null-terminated UTF8. Data segments represent
-initialized data that is loaded directly from the binary into the linear memory
-when the program starts (see [modules](Modules.md#linear-memory-section)).
-
-## Serialized AST
-
-* Use a preorder encoding of the AST
-  * Efficient single-pass validation+compilation and polyfill
-* The data of a node (if there is any), is written immediately after the operator and before child nodes
-  * The operator statically determines what follows, so no generic metadata is necessary.
+This document describes the [portable](Portability.md) binary encoding of the WebAssembly modules.
+
+The binary encoding is a dense representation of module information that enables 
+small files, fast decoding, and reduced memory usage.
+See the [rationale document](Rationale.md#why-a-binary-encoding) for more detail.
+
+The encoding is split into three layers:
+
+* **Layer 0** is a simple pre-order encoding of the AST and related data structures.
+  The encoding is dense and trivial to interact with, making it suitable for
+  scenarios like JIT, instrumentation tools, and debugging.
+* **Layer 1** provides structural compression on top of layer 0, exploiting
```

Comment:
same question as in previous pull - this change removes the text "this is not meant to be standardized currently, but we'll evaluate later". was that intentional? i.e., are you suggesting that the MVP include layer 1?